src/hooks/use-add-site.ts

```diff
@@ -23,6 +27,20 @@ export function useAddSite() {
 		[ sites ]
 	);
 
+	const handleCustomDomainChange = useCallback(
+		( value: string ) => {
+			setCustomDomain( value );
+			// Validate custom domain if enabled
+			const domainPattern = /^[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9](?:\.[a-zA-Z]{2,})+$/;
```

Comment:
```suggestion
			const domainPattern = /^[a-z0-9][a-z0-9-]{0,61}[a-z0-9](?:\.[a-z]{2,})+$/i;
```
A few comments on the validation:
- We could shorten it by adding the case insensitive flag, no?
- It would be appropriate to validate that the total length is less than 253 characters.
- Every domain label can technically contain hyphens. This feels like nitpicking, but we might as well give it a shot.

Reply:
> Every domain label can technically contain hyphens. This feels like nitpicking, but we might as well give it a shot.

I don't understand this part of the comment.

Reply:
For clarity, in the domain name `hello.co.uk`, there are three **labels**: `hello`, `co`, and `uk`.

What I meant by my comment was that the regex currently only allows for hyphens in the first label, while it's technically OK to add them in any label. So, `my-domain.local-network` is technically a valid domain.

I know this is slightly nitpicky, but I think the regex needs relatively little adjustment to make this work.

Reply:
I think the regex allows that now.